### DISCUSSION TITLE
entity data model

### DISCUSSION THREAD

**Discussion by samaremad** [serenity-is/Serenity#5362] [Help]
hello everyone can you help me please

i create data model to manipulate the database in the functions that responsible to do (CRUD) in the controller  files but i found that not all the attribute found 

to be specify i want when the create function execute it will check a boolean field in (user- table ) if it is true then take the id of the user Unit  (which it is the foreign key ) and go to( Units table) increase the( field of votes ) by 1

![Screenshot (35)](https://user-images.githubusercontent.com/77009503/105000735-5faff400-5a37-11eb-81fa-f47d3b21bc92.png)

**Comment by edwardch** [comment]
Take a look in the UserRepository for some inspiration.

Inside your `MySaveHandler` you can override your `AfterSave`.

There are some examples of updating other tables, specifically in the MyDeleteHandler, which [overrides the `OnBeforeDelete` here](https://github.com/serenity-is/Serene/blob/8f0f778d1331d3988610854e0a85e14dbff20757/Serene/Serene.Core/Modules/Administration/User/UserRepository.cs#L277)

**Reply by samaremad** [reply]
first I didn't understand but i discovered that there is missing library ( using System.Linq;)
but my problem now i dont know how to update the table with increment

  var result = new MyRepository().Create(uow, request);



            var n = request.Entity.GroupId;
           var x = request.Entity.Situation;
            var c = db.Units.Where(c => c.group_id == n);
            
            if(x==true)
               

**### (here i want to write that db.units.no_votes +1 )**
            

            
            return result;

**Reply by edwardch** [reply]
In your UserRepository.cs, try something like this
```cs
private class MySaveHandler : SaveRequestHandler<MyRow> 
{
    protected override void AfterSave()
    {
        base.AfterSave();

        new SqlUpdate(YourOtherTableRow.Fields.TableName)
            .Where(YourOtherTableRow.Fields.UserId == Row.UserId.Value)
            .Inc(YourOtherTableRow.Fields.Units, 1)
            .Execute(Connection);
    }
}
```
